tailscale/resource_contacts.go

```diff
@@ -0,0 +1,187 @@
+						"email": {
+							Type:        schema.TypeString,
+							Description: "Email address to send communications to",
+							Required:    true,
+						},
+					},
+				},
+			},
+			"security": {
+				Type:        schema.TypeSet,
+				Description: "Configuration for communications about security issues affecting your tailnet",
+				Required:    true,
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"email": {
+							Type:        schema.TypeString,
+							Description: "Email address to send communications to",
+							Required:    true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceContactsCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*tailscale.Client)
+	account := d.Get("account").(*schema.Set).List()
+	support := d.Get("support").(*schema.Set).List()
+	security := d.Get("security").(*schema.Set).List()
+
+	accountEmail := account[0].(map[string]interface{})["email"].(string)
+	supportEmail := support[0].(map[string]interface{})["email"].(string)
+	securityEmail := security[0].(map[string]interface{})["email"].(string)
+
+	if err := client.UpdateContact(ctx, tailscale.ContactAccount, tailscale.UpdateContactRequest{Email: &accountEmail}); err != nil {
+		return diagnosticsError(err, "Failed to create contacts")
+	}
+
+	if err := client.UpdateContact(ctx, tailscale.ContactSupport, tailscale.UpdateContactRequest{Email: &supportEmail}); err != nil {
+		return diagnosticsError(err, "Failed to create contacts")
+	}
+
+	if err := client.UpdateContact(ctx, tailscale.ContactSecurity, tailscale.UpdateContactRequest{Email: &securityEmail}); err != nil {
+		return diagnosticsError(err, "Failed to create contacts")
+	}
+
+	d.SetId(createUUID())
+	return resourceContactsRead(ctx, d, m)
+}
+
+func resourceContactsRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*tailscale.Client)
+
+	contacts, err := client.Contacts(ctx)
+	if err != nil {
+		return diagnosticsError(err, "Failed to fetch contacts")
+	}
+
+	accountMap := []map[string]interface{}{
+		{
+			"email": contacts.Account.Email,
+		},
+	}
+	if err = d.Set("account", accountMap); err != nil {
+		return diagnosticsError(err, "Failed to set account field")
+	}
+
+	supportMap := []map[string]interface{}{
+		{
+			"email": contacts.Support.Email,
+		},
+	}
+	if err = d.Set("support", supportMap); err != nil {
+		return diagnosticsError(err, "Failed to set support field")
+	}
+
+	securityMap := []map[string]interface{}{
+		{
+			"email": contacts.Security.Email,
+		},
+	}
+	if err = d.Set("security", securityMap); err != nil {
+		return diagnosticsError(err, "Failed to set security field")
+	}
+
+	return nil
+}
+
+func resourceContactsUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*tailscale.Client)
+
+	if d.HasChange("account") {
+		account := d.Get("account").(*schema.Set).List()
```

Comment:
These sections feel a little repetitive, maybe they could be pulled into a helper function?

Reply:
Made an `updateContact` helper to encapsulate some of the repetitive bits here.